lib/trimMessyHeaders.js

```diff
@@ -0,0 +1,6 @@
+module.exports = function trimMessyHeaders(headers) {
+  delete headers.valuesByName['content-length'];
```

Comment:
Couldn't you use `headers.remove('Content-Length');` here? (I know messy isn't documented, so that one is probably on me :)

Reply:
@papandreou hm possibly - I think the body of this function dates from a time where we still had some issues in casing differences, and I didn't want to take any chances about which representation we were using. Stuck with this here because I knew it worked :)

Reply:
`Headers#valuesByName` was meant to be a private property TBH :)